app/src/main/proto/vss.proto

```diff
@@ -0,0 +1,182 @@
+    Authorization and billing can also be performed at the storeId level.
+ */
+  string storeId = 1;
+  /*
+    If present, the write will only succeed if the current server-side globalVersion against
+    a storeId is same as in the request.
+    Clients are expected to store (client-side) the global version against storeId.
+    And the request should contain their client-side value of globalVersion.
+
+    For the first write of the store, global version should be '0'. If the write succeeds, clients
+    must increment their global version (client-side) by 1.
+    The server increments globalVersion (server-side) for every successful write, hence this
+    client-side increment is required to ensure matching versions. This updated global version
+    should be used in subsequent PutObjectRequests for the store.
+
+    Requests with conflicting version will fail with `ConflictException` as ErrorCode.
+   */
+  optional int64 globalVersion = 2;
+
+  /*
+    Items to be written as a result of this PutObjectRequest.
+
+    In an item, each key is supplied with its corresponding value and version.
+    Clients can choose to encrypt the keys client-side in order to obfuscate user's usage patterns.
+    If write is successful, previous value corresponding to the key will be overwritten.
+
+    Multiple items in transactionItems of a single PutObjectRequest are written in
+    a database-transaction in all-or-nothing fashion.
+    Items in single PutObjectRequest must have distinct keys.
+
+    Write will only succeed if current DB version against key is same as in request.
+    Client is expected to store version against every key.
+    When initiating a PutObjectRequest, the request should contain their client-side version for
+    that key-value.
+
+    For first write of key, version should be '0'. If the write succeeds, clients must increment
+    their corresponding key versions (client-side) by 1.
+    The server increments key version (server-side) for every successful write, hence this
+    client-side increment is required to ensure matching versions. These updated key versions should
+    be used in subsequent PutObjectRequests for the keys.
+
+    Requests with conflicting version will fail with `ConflictException` as ErrorCode.
+
+    Considerations for transactions:
+    Transaction writes of multiple items have a performance overhead, hence it is recommended to use
+    them only if required by the client application to ensure logic/code correctness.
+    That is, transactionItems are not a substitute for batch-write of multiple unrelated items.
+    When write of multiple unrelated items is desired, it is recommended to use separate
+    PutObjectRequests.
+
+    Consistency Guarantee:
+    All PutObjectRequests are strongly consistent i.e. they provide read-after-write and
+    read-after-update consistency guarantees.
+   */
+  repeated KeyValue transactionItems = 3;
+}
+
+message PutObjectResponse {
+}
+
```

Comment:
nit: remove extra blank line

Likewise between other messages

Reply:
i was following request-response separated by single blank line and other proto's by double vertical space.
replaced by single blank lines for consistency.